c7n/resources/sagemaker.py

```diff
@@ -220,6 +220,82 @@ def resources(self, query=None):
         return super(SagemakerAutoMLJob, self).resources(query=query)
 
 
+class SagemakerCompilationJobDescribe(DescribeSource):
+
+    def augment(self, resources):
+        return universal_augment(self.manager, super().augment(resources))
+
+
+@resources.register('sagemaker-compilation-job')
+class SagemakerCompilationJob(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'sagemaker'
+        enum_spec = ('list_compilation_jobs', 'CompilationJobSummaries', None)
+        detail_spec = (
+            'describe_compilation_job', 'CompilationJobName', 'CompilationJobName', None)
+        arn = id = 'CompilationJobArn'
+        name = 'CompilationJobName'
+        date = 'CreationTime'
+        permission_prefix = 'sagemaker'
+        universal_taggable = object()
+
+    source_mapping = {'describe': SagemakerCompilationJobDescribe}
+
+    def __init__(self, ctx, data):
+        super(SagemakerCompilationJob, self).__init__(ctx, data)
+        self.queries = QueryFilter.parse(
+            self.data.get('query', [
+                {'StatusEquals': 'INPROGRESS'}]))
+
+    def resources(self, query=None):
+        for q in self.queries:
+            if q is None:
+                continue
+            query = query or {}
+            for k, v in q.items():
+                query[k] = v
+        return super(SagemakerCompilationJob, self).resources(query=query)
+
+
+class SagemakerProcessingJobDescribe(DescribeSource):
+
+    def augment(self, resources):
+        return universal_augment(self.manager, super().augment(resources))
+
+
+@resources.register('sagemaker-processing-job')
+class SagemakerProcessingJob(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'sagemaker'
+        enum_spec = ('list_processing_jobs', 'ProcessingJobSummaries', None)
+        detail_spec = (
+            'describe_processing_job', 'ProcessingJobName', 'ProcessingJobName', None)
+        arn = id = 'ProcessingJobArn'
+        name = 'ProcessingJobName'
+        date = 'CreationTime'
+        permission_prefix = 'sagemaker'
+        universal_taggable = object()
+
+    source_mapping = {'describe': SagemakerProcessingJobDescribe}
+
+    def __init__(self, ctx, data):
+        super(SagemakerProcessingJob, self).__init__(ctx, data)
+        self.queries = QueryFilter.parse(
+            self.data.get('query', [
+                {'StatusEquals': 'InProgress'}]))
+
+    def resources(self, query=None):
+        for q in self.queries:
```

Comment:
this block feels duplicated a lot, feels like its ripe for being extracted to a utility function.

Reply:
separate pr though in that context.

Reply:
filed as https://github.com/cloud-custodian/cloud-custodian/issues/9469